### ISSUE TITLE
Provide derives for `Neg` and the `*Assign` traits.

### ISSUE THREAD

**Issue by oli-cosmian** [rust-num/num-derive#38]
These are equally straight forward as the other numeric impls, I'm guessing there just wasn't any need for these derives so far?

**Comment by cuviper** [comment]
Yeah, it's kind of weird because we don't really have `derive(Add)` etc., just `derive(NumOps)` which really implies the necessary ops. At least `NumOps` is specific to `num-traits`, but maybe we could also break out derives for all of the individual ops. It's a little bit of scope creep, but still in the general `num` domain.
